Could you also include tests in this folder as well? https://github.com/appcelerator/titanium_mobile/tree/master/ti_mocha_tests

Also with regards to this using the test case:-
```javascript
var win = Ti.UI.createWindow({
    backgroundColor : '#ffffff'
});
var button = Ti.UI.createButton({
    top : 10,
    width : 200,
    height : 50,
    title : "write file"
});
win.add(button);
button.addEventListener('click', function(e) {
    var f = Ti.Filesystem.getFile(Ti.Filesystem.applicationDataDirectory, 'fileter.txt');
    f.write('writing to the file would be enough to create it');
    alert(f.read());
});
 
win.open();
```

This is fine.

If I changed the directory to 
```javascript
var f = Ti.Filesystem.getFile(Ti.Filesystem.externalStorageDirectory, 'fileter.txt');
```
I will get the error (On Android 6.0 and above with no permission granted)
```
[ERROR] :  TiFileProxy: (KrollRuntimeThread) [8580,8580] IOException encountered
[ERROR] :  TiFileProxy: java.io.FileNotFoundException: /storage/emulated/0/sg.com.appcelerator/fileter.txt: open failed: EACCES (Permission denied)
```
Is this okay? Should the developer know that externalStorageDirectory requires permission in order to do this or should there be a way to handle this too?

--- a/ti_mocha_tests/ti.filesystem.test.js
+++ b/ti_mocha_tests/ti.filesystem.test.js
@@ -432,9 +432,9 @@ describe("filesystem", function() {
 		finish();
 	});
 	//TIMOB-20440
-	it(“applicationDataDirectory”, function(finish) {
+	it("applicationDataDirectory", function(finish) {
 		var file = Ti.Filesystem.getFile(Ti.Filesystem.applicationDataDirectory, 'filter.txt');
-		var text = “testing a file”;
+		var text = "testing a file";
 		file.write(text);
 		should(file.exists()).be.true;
 		var blob = file.read();

--- a/ti_mocha_tests/ti.filesystem.test.js
+++ b/ti_mocha_tests/ti.filesystem.test.js
@@ -431,6 +431,23 @@ describe("filesystem", function() {
 		}
 		finish();
 	});
+	//TIMOB-20440
+	it(“applicationDataDirectory”, function(finish) {
+		var file = Ti.Filesystem.getFile(Ti.Filesystem.applicationDataDirectory, 'filter.txt');
+		var text = “testing a file”;
+		file.write(text);
+		should(file.exists()).be.true;
+		var blob = file.read();
+		should(blob).not.be.null;
+		var readText = blob.text;
+		should(readText).not.be.null;
+		should(readText).not.be.type("undefined");
+		should(readText).be.a.String;
+		should(readText.length).eql(text.length);
+		should(readText).eql(text);
+		file.deleteFile();
+		finish();
+	}
 	//TIMOB-4469
 	it("existsMethod", function(finish) {
 		should(Titanium.Filesystem.getFile(Titanium.Filesystem.resourcesDirectory, "app.js").exists()).be.true;

